pkg/serverless/flush/strategy.go

```diff
@@ -0,0 +1,95 @@
+package flush
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Strategy is deciding whether the data should be flushed or not at the given moment.
+type Strategy interface {
+	String() string
+	ShouldFlush(moment Moment, t time.Time) bool
+}
+
+// Moment represents at which moment we're asking the flush strategy if we
+// should flush or not.
+// Note that there is no entry for the shutdown of the environment because we always
+// flush in this situation.
+type Moment string
+
+const (
+	// Starting is used to represent the moment the function is starting because
+	// it has been invoked.
+	Starting Moment = "starting"
```

Comment:
What about `Started` since starting implies a short time at the start of the function but `Periodically` could flush minutes after this ?

Reply:
It's the proper term, the function is just starting.